### PR TITLE
GRAD2-2639: initial implementation is done.

### PR DESCRIPTION
**Pull request by infstar**
GRAD2-2639: initial implementation is done.

### CODE REVIEW

File: api/src/main/java/ca/bc/gov/educ/api/batchgraduation/config/BatchJobConfig.java [hunk 1]
```diff
@@ -1086,7 +1086,7 @@ public Step schoolReportsRegenJobStep(JobRepository jobRepository, PlatformTrans
 
     @Bean
     public Step masterStepSchoolReportsRegen(JobRepository jobRepository, PlatformTransactionManager transactionManager, EducGradBatchGraduationApiConstants constants, SkipSQLTransactionExceptionsListener skipListener) {
-        int partitionSize = constants.getNumberOfPartitions() / 2;
+        int partitionSize = (constants.getNumberOfPartitions() / 2) + 1;
```
Comment (cditcher) [inline api/src/main/java/ca/bc/gov/educ/api/batchgraduation/config/BatchJobConfig.java:1089]: Why add one here?

File: api/src/main/java/ca/bc/gov/educ/api/batchgraduation/reader/DeleteStudentReportsPartitioner.java [hunk 2]
```diff
@@ -48,13 +48,13 @@ public Map<String, ExecutionContext> partition(int gridSize) {
         logger.debug("Filter Schools for deleting student reports");
         boolean processAllReports = "ALL".equalsIgnoreCase(searchRequest.getActivityCode());
         Long batchId = jobExecution.getId();;
```
Comment (cditcher) [inline api/src/main/java/ca/bc/gov/educ/api/batchgraduation/reader/DeleteStudentReportsPartitioner.java:50]: I know it's not your code, but please remove double ;; on line 50.

**Review by cditcher — CHANGES_REQUESTED** [review]

File: api/src/main/java/ca/bc/gov/educ/api/batchgraduation/config/BatchJobConfig.java (hunk 1 above)
Comment (infstar) [reply api/src/main/java/ca/bc/gov/educ/api/batchgraduation/config/BatchJobConfig.java:1089]: Oops, that's my local stuff for debugging.   I will revert this.

File: api/src/main/java/ca/bc/gov/educ/api/batchgraduation/reader/DeleteStudentReportsPartitioner.java (hunk 2 above)
Comment (infstar) [reply api/src/main/java/ca/bc/gov/educ/api/batchgraduation/reader/DeleteStudentReportsPartitioner.java:50]: I will remove the double.

File: api/src/main/java/ca/bc/gov/educ/api/batchgraduation/config/BatchJobConfig.java (hunk 1 above)
Comment (infstar) [reply api/src/main/java/ca/bc/gov/educ/api/batchgraduation/config/BatchJobConfig.java]: Done

File: api/src/main/java/ca/bc/gov/educ/api/batchgraduation/reader/DeleteStudentReportsPartitioner.java (hunk 2 above)
Comment (infstar) [reply api/src/main/java/ca/bc/gov/educ/api/batchgraduation/reader/DeleteStudentReportsPartitioner.java]: Done

**Review by cditcher — APPROVED** [review]

**Pull request merged**